File: phant/flashUp.js
#!/usr/bin/env node
// Pings google until it gets a response and then beeps.
// apt-get install flite

var child = require('child_process');
var b = require('bonescript');

var pingCmd = "ping -w1 google.com";
var ms = 1500;  // Repeat time.
var red   = 'P9_11';
var green = 'P9_15';
var blue  = 'P9_13';

b.pinMode(red,   b.OUTPUT);
b.pinMode(green, b.OUTPUT);
b.pinMode(blue,  b.OUTPUT);

b.digitalWrite(red,   0);
b.digitalWrite(green, 0);
b.digitalWrite(blue,  0);


// console.log("process.argv.length: " + process.argv.length);
if(process.argv.length === 3) {
    pingCmd = process.argv[2];
}

var timer = setInterval(ping, ms);

// Send off the ping command.
function ping  () {
    child.exec(pingCmd,
        function (error, stdout, stderr) {
            if(error || stderr) { 
                console.log('error: ' + error); 
                console.log('stderr: ' + stderr); 
                b.digitalWrite(red,   1);
                b.digitalWrite(green, 0);
            } else {
                console.log('ping: ' + stdout);
                b.digitalWrite(red,   0);
                b.digitalWrite(green, 1);
            }
        }
    )
}

File: iot/google/sheets/flashUp.py
#!/usr/bin/env python3
# Pings google and turns on a RGB light according to the response time.

import gpiod
import subprocess
import time
import re
import datetime as dt

pingCmd = ["ping", "-w1", "www.google.com"]
p = re.compile('time=[0-9.]*')      # We'll search for time= followed by digits and .
ms = 15000          # Repeat time in ms.
thresh = 40.0      # If time is above this, turn on warning light
hist = 10* [thresh]  # Keep track of the last 10 values
current = 0        # Insert next value here 

LED_CHIP = 'gpiochip0'
red   = 30 # 'P9_11'
green =  5 # 'P9_17'
blue  = 31 # 'P9_13'
LED_LINE_OFFSET = [red, green, blue]

chip = gpiod.Chip(LED_CHIP)
lines = chip.get_lines(LED_LINE_OFFSET)
lines.request(consumer='flashup.py', type=gpiod.LINE_REQ_DIR_OUT)

# Turn all LEDs off
def allOff():
    print("allOff")
    lines.set_values([0, 0, 0])  # All off

lines.set_values([1, 0, 1])     # red and blue

while True:
    hour=dt.datetime.now().hour
    if hour>5 and hour<21:
        try:
            # returns output as byte string
            returned_output = subprocess.check_output(pingCmd, stderr=subprocess.STDOUT).decode("utf-8")
            # print("returned_output: {}".format(returned_output))
            
            # using decode() function to convert byte string to string
            # print('{}:\n{}'.format(pingCmd, returned_output))
            timems = float(p.search(returned_output).group()[5:])
            # print(timems)
            average = sum(hist)/len(hist)
            hist[current] = timems
            current = current + 1
            if current >= len(hist):
                current=0
            print('ping: time = %5.2f, average = %5.2f' % (timems, average))
            if timems > 1.1*average:
                lines.set_values([1, 1, 0])     # red and green
            else:
                lines.set_values([0, 1, 0])     # green

                
        except subprocess.CalledProcessError as err:
            print('ERROR:', err)
            lines.set_values([1, 0, 0])#        # red

            
    else:
        print('Good night')
        allOff()
    time.sleep(ms/1000)

# process.on('SIGINT', function() {
#     console.log('Got SIGINT');
#     clearInterval(timer);
#     setTimeout(allOff, 1000);
# });
